test/tools/Modules/PSSysLog/PSSysLog.psm1

```diff
@@ -0,0 +1,797 @@
+
+3: Run 'log collect' to retrieve log records after a specific timestamp
+from the system logs.
+Note that the extracted data is still not directly consumable.
+
+4: Run 'log show' to convert the extracted data to text and redirect it
+to a file.
+The --predicate can be used at this point to filter extracted records.
+The typical filter is 'process == "pwsh"'
+
+The redirected text file can then be consumed by this module.
+
+Example command-lines:
+
+sudo log collect --start "2018-02-07 14:33:30" --output ./system.logarchive
+log show ./system.logarchive/ --info --predicate 'process == "pwsh"' >pwsh.log.txt
+
+Parsing Notes:
+1: Sample contains 6.0.1 content (which is out of date) revise with 6.1.0 preview
+2: Ensure analytic data is considered when parsing; specifically Provider_Lifecycle:ProviderStart.Method.Informational
+3: Multi-line output is expected. Parsing needs to detect the timestamp at the begining
+of a line and append subsequent lines to the message until the next 'log' like is found.
+4: Header lines need to be skipped.
+
+Sample output from 'log show' illustrating one single-line entry and one multi-line entry.
+Analytic log items are often multi-line because the message text contains newline characters.
+
+==========
+/Users/psbuildacct/system.logarchive
+==========
+Timestamp                       Thread     Type        Activity             PID
+2018-02-07 14:34:35.256501-0800 0x2a3730   Default     0x0                  39437  pwsh: (libpsl-native.dylib) [com.microsoft.powershell.powershell] (v6.0.1:1:10) [Perftrack_ConsoleStartupStart:PowershellConsoleStartup.WinStart.Informational] PowerShell console is starting up
+2018-02-07 14:34:35.562003-0800 0x2a373a   Default     0x0                  39437  pwsh: (libpsl-native.dylib) [com.microsoft.powershell.powershell] (v6.0.1:4:11) [Provider_Lifecycle:ProviderStart.Method.Informational] Provider Alias changed state to Started.
+Context:
+        Severity = Informational
+        Host Name = ConsoleHost
+        Host Version = 6.0.1
+        Host ID = 964756e8-c074-4228-a54d-d410a88e8c66
+        Host Application = /usr/local/microsoft/powershell/6.0.1/pwsh.dll
+        Engine Version =
+        Runspace ID =
+        Pipeline ID =
+        Command Name =
+        Command Type =
+        Script Name =
+        Command Path =
+        Sequence Number = 1
+        User = PowerShells-MacBook\psbuildacct
+        Connected User =
+        Shell ID = Microsoft.PowerShell
+        User Data:
+#>
+
+#region Utilities
+function Test-Elevated
+{
+    [CmdletBinding()]
+    [OutputType([bool])]
+    Param()
+
+    # if the current Powershell session was called with administrator privileges,
+    # the Administrator Group's well-known SID will show up in the Groups for the current identity.
+    # Note that the SID won't show up unless the process is elevated.
+    return (([Security.Principal.WindowsIdentity]::GetCurrent()).Groups -contains "S-1-5-32-544")
```

Comment:
should this be protected by try/catch?
`try { ... } catch { return $false }`

Reply:
That's a placeholder - WIP

Reply:
@JamesWTruher: that code has been replaced.